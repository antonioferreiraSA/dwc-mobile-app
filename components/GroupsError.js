import React from 'react';
import { StyleSheet, View } from 'react-native';
import { Subtitle } from '../components/shared/Typography';
import Button from './shared/Button';

// eslint-disable-next-line
export default ({ tryAgain }) => {
  const hasTryAgain =
    tryAgain && {}.toString.call(tryAgain) === '[object Function]';

  return (
    <View style={styles.container}>
      <Subtitle center style={styles.title}>
        Something went wrong...
      </Subtitle>
      {hasTryAgain && <Button title="Try Again" onPress={() => tryAgain()} />}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 40,
    justifyContent: 'center',
  },
  title: { marginBottom: 20 },
});
