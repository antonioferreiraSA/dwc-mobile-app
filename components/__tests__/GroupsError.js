import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import GroupsError from '../GroupsError';

describe('<GroupsError/>', () => {
  test('render a group error', async () => {
    const { toJSON } = render(<GroupsError />);

    expect(toJSON()).toMatchSnapshot();
  });

  test('render a group error with a Try Again CTA', async () => {
    const mockTryAgain = jest.fn();
    const { toJSON, getByText } = render(
      <GroupsError tryAgain={mockTryAgain} />
    );

    expect(toJSON()).toMatchSnapshot();

    fireEvent.press(getByText('Try Again'));
    expect(mockTryAgain).toBeCalled();
  });
});
