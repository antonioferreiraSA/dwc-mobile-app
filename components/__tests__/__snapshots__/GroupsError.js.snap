// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupsError/> render a group error 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
      "padding": 40,
    }
  }
>
  <Text
    center={true}
    selectable={true}
    style={
      Array [
        Object {
          "color": "#cbd2d6",
          "fontFamily": "NunitoSans-Regular",
          "fontSize": 18,
          "lineHeight": 24,
          "textAlign": "center",
        },
        Array [
          Object {
            "color": "#cbd2d6",
            "fontSize": 22,
            "lineHeight": 28,
            "marginBottom": 20,
          },
          Object {
            "marginBottom": 20,
          },
        ],
      ]
    }
  >
    Something went wrong...
  </Text>
</View>
`;

exports[`<GroupsError/> render a group error with a Try Again CTA 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
      "padding": 40,
    }
  }
>
  <Text
    center={true}
    selectable={true}
    style={
      Array [
        Object {
          "color": "#cbd2d6",
          "fontFamily": "NunitoSans-Regular",
          "fontSize": 18,
          "lineHeight": 24,
          "textAlign": "center",
        },
        Array [
          Object {
            "color": "#cbd2d6",
            "fontSize": 22,
            "lineHeight": 28,
            "marginBottom": 20,
          },
          Object {
            "marginBottom": 20,
          },
        ],
      ]
    }
  >
    Something went wrong...
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#313d48",
          "borderRadius": 30,
          "justifyContent": "center",
          "minHeight": 60,
          "paddingHorizontal": 16,
          "paddingVertical": 12,
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        selectable={true}
        style={
          Array [
            Object {
              "color": "#cbd2d6",
              "fontFamily": "NunitoSans-Regular",
              "fontSize": 18,
              "lineHeight": 24,
              "textAlign": "auto",
            },
            Array [
              Object {
                "color": "#fff",
                "textAlign": "center",
              },
              Object {},
            ],
          ]
        }
      >
        Try Again
      </Text>
    </View>
  </View>
</View>
`;
